### PR TITLE
Security Considerations: Added text on Ed25519 security properties.

### PR DESCRIPTION
**Pull request by Wind4Greg**
Based on discussions concerning issue https://github.com/w3c/vc-di-eddsa/issues/45 I have come up with some *informative* text for the `security considerations` section. This text explains the relatively recent work that defines and proves security properties for Ed25519 signatures, how to achieve the highest levels of security, and cautions on Ed25519 library implementations.

Note that I didn't see an "easy way" to include mathematical notation with ReSpec so used MathML (a W3C specification) which is somewhat verbose and not hand editable. Some online math editors such as [LaTex Editor](https://latexeditor.lagrida.com/) can produce MathML output if you want to edit any of the equations.

Cheers Greg

***
<a href="https://pr-preview.s3.amazonaws.com/Wind4Greg/vc-di-eddsa/pull/47.html" title="Last updated on Jun 1, 2023, 6:24 PM UTC (ea626af)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-di-eddsa/47/8031fd9...Wind4Greg:ea626af.html" title="Last updated on Jun 1, 2023, 6:24 PM UTC (ea626af)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -729,6 +744,561 @@ <h2>Security Considerations</h2>
 mitigate the attacks.
       </p>
 
+      <section class="informative">
+        <h3>Security Properties of Ed25519 Implementations</h3>
+        <p>There has 
+been wide adoption of Ed25519 signatures (EdDSA algorithm with edwards25519 
+curve) due to the compact size of the keys and signatures, and the speed at
```
Comment (TallTed) [inline index.html]:
```suggestion
        <p>Ed25519 signatures (EdDSA algorithm with edwards25519 curve) have
        been widely adopted, due both to the compact size of the keys and
        signatures and to the speed at 
```

File: index.html [hunk 2]
```diff
@@ -729,6 +744,561 @@ <h2>Security Considerations</h2>
 mitigate the attacks.
       </p>
 
+      <section class="informative">
+        <h3>Security Properties of Ed25519 Implementations</h3>
+        <p>There has 
+been wide adoption of Ed25519 signatures (EdDSA algorithm with edwards25519 
+curve) due to the compact size of the keys and signatures, and the speed at 
+which signatures can be produced and verified. Many libraries exist that can
+create and verify Ed25519 signatures. Since the publication of [[RFC8032]]
```
Comment (TallTed) [inline index.html]:
```suggestion
create and verify Ed25519 signatures. Since the publication of [[RFC8032]],
```

File: index.html [hunk 3]
```diff
@@ -729,6 +744,561 @@ <h2>Security Considerations</h2>
...
 
+      <section class="informative">
+        <h3>Security Properties of Ed25519 Implementations</h3>
+        <p>There has 
+been wide adoption of Ed25519 signatures (EdDSA algorithm with edwards25519 
+curve) due to the compact size of the keys and signatures, and the speed at 
+which signatures can be produced and verified. Many libraries exist that can
+create and verify Ed25519 signatures. Since the publication of [[RFC8032]] 
+security properties of Ed25519 signatures have been rigorously proven 
+[[Provable_Ed25519]] and [[Taming_EdDSAs]]. However it was observed that a
```
Comment (TallTed) [inline index.html]:
```suggestion
security properties of Ed25519 signatures have been rigorously proven (see
[[Provable_Ed25519]] and [[Taming_EdDSAs]]). However, it has been observed that a
```

File: index.html [hunk 4]
```diff
@@ -729,6 +744,561 @@ <h2>Security Considerations</h2>
...
+      <section class="informative">
+        <h3>Security Properties of Ed25519 Implementations</h3>
+        <p>There has 
+been wide adoption of Ed25519 signatures (EdDSA algorithm with edwards25519 
+curve) due to the compact size of the keys and signatures, and the speed at 
+which signatures can be produced and verified. Many libraries exist that can
+create and verify Ed25519 signatures. Since the publication of [[RFC8032]] 
+security properties of Ed25519 signatures have been rigorously proven 
+[[Provable_Ed25519]] and [[Taming_EdDSAs]]. However it was observed that a 
+significant number of libraries do not achieve these security properties due to
```
Comment (TallTed) [inline index.html]:
```suggestion
significant number of libraries do not achieve these security levels, due to 
```

File: index.html [hunk 5]
```diff
@@ -729,6 +744,561 @@ <h2>Security Considerations</h2>
...
+curve) due to the compact size of the keys and signatures, and the speed at 
+which signatures can be produced and verified. Many libraries exist that can
+create and verify Ed25519 signatures. Since the publication of [[RFC8032]] 
+security properties of Ed25519 signatures have been rigorously proven 
+[[Provable_Ed25519]] and [[Taming_EdDSAs]]. However it was observed that a 
+significant number of libraries do not achieve these security properties due to 
+missing input validity checks during the signature verification process. In 
+this section we summarize the security properties achievable with Ed25519 
+signatures and indicate how one can determine if a library will support those 
+properties.
```
Comment (TallTed) [inline index.html]:
```suggestion
this section, we summarize the security levels achievable with Ed25519 
signatures, and indicate how one can determine whether a library will support those 
levels.
```

**Review by TallTed — CHANGES_REQUESTED** [review]

**Review by msporny — APPROVED** [review]
LGTM. Bonus points for using MathML! :)

**Comment by msporny** [comment]
Editorial, multiple reviews, changes requested and made, no objections, merging.

**Pull request merged**